hbase-server/src/test/java/org/apache/hadoop/hbase/master/cleaner/TestCleanerChore.java

```diff
@@ -154,16 +160,25 @@ public FileStatus[] listStatus(Path f) throws IOException {
 
       // filesystem is back
       fails.set(false);
-      for (;;) {
-        CompletableFuture<Boolean> succFuture = chore.triggerCleanerNow();
-        // the reset of the future is async, so it is possible that we get the previous future
-        // again.
-        if (succFuture != errorFuture) {
-          // verify that it accurately reported success.
-          assertTrue("chore should claim it succeeded.", succFuture.get());
+      CompletableFuture<Boolean> succFuture = chore.triggerCleanerNow();
+      // all invocations after the signal are expected to succeed. the future handle we have
+      // currently may or may succeed. an entire run might be scheduled between now and when we're
+      // signaled, so we cannot naively check equality of succFuture vs. errorFuture.
+      successSignal.await();
+      for (int i = 0; i < 5; i++) {
+        if (succFuture == null) {
```

Comment:
nit: we could user HBaseCommonTestingUtility.waitFor here

Reply:
+1